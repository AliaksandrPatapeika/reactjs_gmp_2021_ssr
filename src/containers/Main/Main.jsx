import './Main.less';

import {useRouter} from 'next/router';
import React, {useEffect} from 'react';
import {useDispatch, useSelector} from 'react-redux';

import {fetchMovies, setStateFromURL} from '../../actions/movies';
import Menu from '../Menu';
import MoviesList from '../MoviesList';
import NoMoviesFound from '../NoMoviesFound';
import ResultCount from '../ResultCount';

const Main = () => {
  const activeModalWindow = useSelector((state) => state.movie.activeModalWindow);
  const search = useSelector((state) => state.query.search);
  const filter = useSelector((state) => state.query.filter);
  const sortBy = useSelector((state) => state.query.sortBy);
  const sortOrder = useSelector((state) => state.query.sortOrder);
  const limit = useSelector((state) => state.query.limit);
  const dispatch = useDispatch();
  const router = useRouter();

  useEffect(() => {
    dispatch(setStateFromURL(router.query));
    dispatch(fetchMovies());
  }, [dispatch, search, filter, sortBy, sortOrder, limit, router.query]);

  return !activeModalWindow && (
  <main className="mainContainer">
    <Menu />
    <ResultCount />
    <MoviesList />
    <NoMoviesFound />
  </main>
  );
};

export default Main;
