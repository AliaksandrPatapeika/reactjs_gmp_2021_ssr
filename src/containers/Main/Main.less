@import "../../less/variables";

.mainContainer {
	padding: 0 62px;
	background-color: @black;
}
