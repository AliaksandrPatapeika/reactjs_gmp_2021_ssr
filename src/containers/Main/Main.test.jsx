import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import MockMovies from '../../tests/mocks/mockData';
import Main from './Main';

jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {
      search: 'The Movie',
      filter: 'comedy',
      sortBy: 'runtime',
      sortOrder: 'asc',
      limit: 6
    }
  })
}));

describe('Main', () => {
  test('renders correctly', () => {
    const mockStore = configureMockStore();
    const initialState = {
      movie: {
        isLoading: true,
        movies: MockMovies,
        totalAmount: 6
      },
      query: {
        filter: '',
        sortBy: 'release_date',
        sortOrder: 'desc'
      }
    };
    const store = mockStore(initialState);
    const main = renderer
      .create(
        <Provider store={store}>
          <Main />
        </Provider>
      )
      .toJSON();

    expect(main)
      .toMatchSnapshot();
  });
});
