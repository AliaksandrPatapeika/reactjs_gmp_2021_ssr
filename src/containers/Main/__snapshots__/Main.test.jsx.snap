// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
<main
  className="mainContainer"
>
  <div
    className="menuContainer"
  >
    <nav>
      <ul
        className="resultsFilterContainer"
      >
        <li
          className="filterItem active"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          ALL
        </li>
        <li
          className="filterItem"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          DOCUMENTARY
        </li>
        <li
          className="filterItem"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          COMEDY
        </li>
        <li
          className="filterItem"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          HORROR
        </li>
        <li
          className="filterItem"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          CRIME
        </li>
      </ul>
    </nav>
    <div
      className="resultsSortContainer"
    >
      <div
        className="sortBy"
      >
        SORT BY
      </div>
      <div
        className="select resultSortSelect css-2b097c-container"
        id="resultSort"
        onKeyDown={[Function]}
      >
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="select__control css-yk16xz-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="select__value-container select__value-container--has-value css-g1d714-ValueContainer"
          >
            <div
              className="select__single-value css-1uccc91-singleValue"
            >
              RELEASE DATE
            </div>
            <div
              className="css-b8ldur-Input"
            >
              <div
                className="select__input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-autocomplete="list"
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  disabled={false}
                  id="react-select-resultSort-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "boxSizing": "content-box",
                      "color": "inherit",
                      "fontSize": "inherit",
                      "label": "input",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "1px",
                    }
                  }
                  tabIndex="0"
                  type="text"
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
          </div>
          <div
            className="select__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className="select__indicator-separator css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className="select__indicator select__dropdown-indicator css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <i
                aria-hidden="true"
                className="fa fa-caret-down"
              />
            </div>
          </div>
        </div>
      </div>
      <button
        className="arrow"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <i
          aria-hidden="true"
          className="fa fa-sort-amount-desc"
        />
      </button>
    </div>
  </div>
  <div
    className="resultCountContainer"
  >
    <span
      className="resultCount"
    >
      6
       
    </span>
    <span>
      movies found
    </span>
  </div>
  <div
    className="moviesListContainer"
  >
    <div
      className="movieCardContainer"
      onMouseLeave={[Function]}
    >
      <button
        className="posterButton"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="movie card poster"
          className="movieCardPoster"
          onError={[Function]}
          src="http://localhost:3000/img/01-pulp-fiction.jpg"
        />
      </button>
      <button
        className="threeDotsIcon"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="movie card menu"
          src="/img/moreButton.svg"
        />
      </button>
      <div
        className="movieTitleReleaseDateContainer"
      >
        <div
          className="movieTitle"
        >
          Pulp Fiction
        </div>
        <div
          className="movieReleaseDate"
          title="Release date: 2004-03-15"
        >
          2004
        </div>
      </div>
      <div
        className="movieGenres"
      >
        Action & Adventure
      </div>
    </div>
    <div
      className="movieCardContainer"
      onMouseLeave={[Function]}
    >
      <button
        className="posterButton"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="movie card poster"
          className="movieCardPoster"
          onError={[Function]}
          src="http://localhost:3000/img/02-bohemian-rhapsody.jpg"
        />
      </button>
      <button
        className="threeDotsIcon"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="movie card menu"
          src="/img/moreButton.svg"
        />
      </button>
      <div
        className="movieTitleReleaseDateContainer"
      >
        <div
          className="movieTitle"
        >
          Bohemian Rhapsody
        </div>
        <div
          className="movieReleaseDate"
          title="Release date: 2003-05-23"
        >
          2003
        </div>
      </div>
      <div
        className="movieGenres"
      >
        Drama, Biography, Music
      </div>
    </div>
    <div
      className="movieCardContainer"
      onMouseLeave={[Function]}
    >
      <button
        className="posterButton"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="movie card poster"
          className="movieCardPoster"
          onError={[Function]}
          src="http://localhost:3000/img/03-kill-bill.jpg"
        />
      </button>
      <button
        className="threeDotsIcon"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="movie card menu"
          src="/img/moreButton.svg"
        />
      </button>
      <div
        className="movieTitleReleaseDateContainer"
      >
        <div
          className="movieTitle"
        >
          Kill Bill: Vol 2
        </div>
        <div
          className="movieReleaseDate"
          title="Release date: 1994-01-08"
        >
          1994
        </div>
      </div>
      <div
        className="movieGenres"
      >
        Oscar winning Movie
      </div>
    </div>
    <div
      className="movieCardContainer"
      onMouseLeave={[Function]}
    >
      <button
        className="posterButton"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="movie card poster"
          className="movieCardPoster"
          onError={[Function]}
          src="http://localhost:3000/img/04-avengers-infinity-war.jpg"
        />
      </button>
      <button
        className="threeDotsIcon"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="movie card menu"
          src="/img/moreButton.svg"
        />
      </button>
      <div
        className="movieTitleReleaseDateContainer"
      >
        <div
          className="movieTitle"
        >
          Avengers: War of Infinity
        </div>
        <div
          className="movieReleaseDate"
          title="Release date: 2004-07-21"
        >
          2004
        </div>
      </div>
      <div
        className="movieGenres"
      >
        Action & Adventure
      </div>
    </div>
    <div
      className="movieCardContainer"
      onMouseLeave={[Function]}
    >
      <button
        className="posterButton"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="movie card poster"
          className="movieCardPoster"
          onError={[Function]}
          src="http://localhost:3000/img/05-inception.jpg"
        />
      </button>
      <button
        className="threeDotsIcon"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="movie card menu"
          src="/img/moreButton.svg"
        />
      </button>
      <div
        className="movieTitleReleaseDateContainer"
      >
        <div
          className="movieTitle"
        >
          Inception
        </div>
        <div
          className="movieReleaseDate"
          title="Release date: 2003-05-02"
        >
          2003
        </div>
      </div>
      <div
        className="movieGenres"
      >
        Action & Adventure
      </div>
    </div>
    <div
      className="movieCardContainer"
      onMouseLeave={[Function]}
    >
      <button
        className="posterButton"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="movie card poster"
          className="movieCardPoster"
          onError={[Function]}
          src="http://localhost:3000/img/06-reservoir-dogs.jpg"
        />
      </button>
      <button
        className="threeDotsIcon"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        <img
          alt="movie card menu"
          src="/img/moreButton.svg"
        />
      </button>
      <div
        className="movieTitleReleaseDateContainer"
      >
        <div
          className="movieTitle"
        >
          Reservoir dogs
        </div>
        <div
          className="movieReleaseDate"
          title="Release date: 1994-11-30"
        >
          1994
        </div>
      </div>
      <div
        className="movieGenres"
      >
        Oscar winning Movie
      </div>
    </div>
  </div>
</main>
`;
